src/main/java/frc/robot/commands/auton/AutonBuilder.java

```diff
@@ -0,0 +1,280 @@
+            xPID, 
+            yPID, 
+            thetaController, 
+            ((ChassisSpeeds speeds) -> {
+                swerveSubsystem.setChassisSpeeds(
+                    speeds.vxMetersPerSecond,
+                    speeds.vyMetersPerSecond, 
+                    speeds.omegaRadiansPerSecond
+                );
+            }),
+            fmsSubsystem::isRedAlliance,
+            swerveSubsystem
+        );
+    }
+
+    /**
+     * Follows trajectory to intake.
+     *
+     * @param intakeTrajectory ChoreoTrajectory
+     * @return goIntake Command
+     */
+    public Command goIntake(ChoreoTrajectory intakeTrajectory) {
+        return followPath(intakeTrajectory).alongWith(
+            new ElevatorToIntakeCommand(elevatorSubsystem).andThen(
+                new IntakePivotSetPositionCommand(intakePivotSubsystem, 1)
+            )
+        ).andThen(
+            new IntakeRollerIntakeCommand(intakeRollerSubsystem, lightBarSubsystem)
+                .raceWith(new DriveForwardCommand(swerveSubsystem).withTimeout(AutonConstants.INTAKE_SWERVE_TIME)),
+            new IntakePivotSetPositionCommand(intakePivotSubsystem, 0)
+        );
+    }
+
+    /** 
+     * Shoots at calculated robot angle and shooter angle.
+     *
+     * @return shoot command
+     */
+    public Command shoot() {
+        return new ShooterPivotAimCommand(shooterPivotSubsystem)
+            .alongWith(new SetCalculatedAngleCommand(swerveSubsystem))
+            .andThen(new IntakeRollerFeedCommand(intakeRollerSubsystem).withTimeout(AutonConstants.SHOOT_FEED_TIME)
+        );
+    }
+
+    /**
+     * Follows trajectory and then shoots at calculated robot angle and shooter angle.
+     *
+     * @param shootTrajectory ChoreoTrajectory
+     * @return goShoot command
+     */
+    public SequentialCommandGroup goShoot(ChoreoTrajectory shootTrajectory) {
+        return followPath(shootTrajectory)
+        .andThen(shoot());
+    }
+
+    /**
+     * Resets the swerve pose estimation to a newPose.
+     *
+     * @param newPose The pose to reset pose estimation to.
+     * @return The command to reset the pose.
+     */
+    public InstantCommand resetSwerve(Pose2d newPose) {
+        return new InstantCommand(() -> swerveSubsystem.resetPose(newPose), swerveSubsystem);
+    }
+
+    /**
+     * Wraps auton commands with common auton commands.
+     *
+     * @param initPose The initial pose to reset swerve to.
+     * @param commands The commands to run during auton.
+     * @return The full auton sequence.
+     */
+    public SequentialCommandGroup buildAuton(Pose2d initPose, Command... commands) {
+        SequentialCommandGroup autonSequence = new SequentialCommandGroup();
+
+        autonSequence.addCommands(
+            resetSwerve(GRTUtil.mirrorAcrossField(initPose, fmsSubsystem::isRedAlliance)),
+            new ShooterFlywheelReadyCommand(shooterFlywheelSubsystem, lightBarSubsystem)
+        );
+        autonSequence.addCommands(commands);
```

Comment:
You *might* be able to get away with
```java
autonSequence.addCommands(
    resetSwerve(...),
    new ShooterFlywheelReadyCommand(...),
    *commands,
    ...
);
```
using Java spread syntax though I don't know if the type system gets icky if you do that.